### PR TITLE
Federico/dw 54/add comments apis

### PR DESCRIPTION
**Pull request by FedericoCampe8**
This commit adds the comments page and real-time APIs to add comments, like comments, etc. The UI is just a mock while the DB queries are working.

New libraries:
- npm install uuid: to get a unique id for each comment (required, for example, to delete it)
- npm install tailwind-scrollbar-hide: not used yet but installed
- npm i safe-json-stringify: not used but installed

### CODE REVIEW

File: components/Comment.js [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+  // function that will set the number of likes on
+  // each change of the DB (triggered by onSnapshot)
+  useEffect(() => {
+    db.collection("posts")
+      .doc(postId)
+      .collection("comments")
+      .doc(commentId)
+      .onSnapshot((snapshot) => {
+        // Get the likes map
+        const likesMap = snapshot.data().likes;
```
Comment (EandrewJones) [inline components/Comment.js:23]: Aren't likes stored as an int? Why are they returned here as a map? It seems inefficient to have to run a for loop to update the likes count every time the state changes.

File: components/Comment.js [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+
+      <p className="text-black mb-2">{comment.message}</p>
+
+      {/* Likes and other stuff go here */}
+      <h2 className="text-black mb-2">
+        {getNumLikes()}
+        Likes
+      </h2>
+
+      {/* Likes: the user can either LIKE or DISLIKE */}
```
Comment (EandrewJones) [inline components/Comment.js:91]: There needs to be a condition here that prevents users from liking their own posts, assuming we don't wan to let users like their own posts.

Comment (EandrewJones) [reply components/Comment.js:91]: I can add  this, but want to get consensus w/ you on this assumption.

**Review by EandrewJones — APPROVED** [review]

File: components/Comment.js (hunk 1 above)
Comment (FedericoCampe8) [reply components/Comment.js:23]: How do you pair likes with users?
Given a post:
User A clicks on like -> 1 vote
User B clicks on like  -> 2 vote
User A clicks on like again -> 1 vote
User A clicks on like again -> 2 votes
User C clicks on like -> 3 votes

If you have 1 int how can you keep track of:

- a user already liked the post and wants to dislike it
- a user likes multiple time the same post

File: components/Comment.js (hunk 2 above)
Comment (FedericoCampe8) [reply components/Comment.js:91]: I think it is fine, I can like my own post on facebook or Discords, it is just 1 like...

File: components/Comment.js (hunk 1 above)
Comment (EandrewJones) [reply components/Comment.js:23]: Got it. I've kept track of likes with their own model in the past which I suppose was the user -> post mapping. So posts and users were dynamically links via likes and you could just perform a count on the number of likes for a given post.

That was with MySQL though and DBs are not my expertise. That many queries could very well be slower than a loop in js.

File: components/Comment.js (hunk 2 above)
Comment (EandrewJones) [reply components/Comment.js:91]: Okay.

**Pull request merged**